var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -476,35 +476,28 @@ def cache_test_sources(self):
         install test subdirectory for use during `spack test run`."""
         self.cache_extra_test_sources([join_path("examples", "local_function_tasks")])
 
-    def run_local_function_tasks_test(self):
-        """Run stand alone test: local_function_tasks"""
+    def test_run_local_function_tasks(self):
+        """Run local_function_tasks"""
 
         test_dir = join_path(
             self.test_suite.current_test_cache_dir, "examples", "local_function_tasks"
         )
 
         if not os.path.exists(test_dir):
-            print("Skipping local_function_tasks test")
-            return
-
-        exe = "local_function_tasks"
+            raise SkipTest(f"{test_dir} must exist")
 
         cmake_args = [
             f"-DCMAKE_C_COMPILER={self.compiler.cc}",
             f"-DCMAKE_CXX_COMPILER={self.compiler.cxx}",
             f"-DLegion_DIR={join_path(self.prefix, 'share', 'Legion', 'cmake')}",
         ]
 
-        self.run_test(
-            "cmake",
-            options=cmake_args,
-            purpose=f"test: generate makefile for {exe} example",
-            work_dir=test_dir,
-        )
-
-        self.run_test("make", purpose=f"test: build {exe} example", work_dir=test_dir)
+        with working_dir(test_dir):
+            cmake = which("cmake")
```

Comment:
Since the `cmake` dependency of this package is for a specific version, it is safer in terms of ensuring the proper `cmake` is being used to also convert this as well:

```suggestion
            cmake = self.spec["cmake"].command
```